modules/material/themes/material/silauth/loginuserpass.twig

```diff
@@ -0,0 +1,146 @@
+<!DOCTYPE html>
+<html lang="{{ currentLanguage }}">
+<head>
+  <title>{{ '{login:title}'|trans({'%idpName%': idpName}) }}</title>
+
+  {% include 'header.twig' %}
+
+  <script>
+    function preventDefault(event) {
+      event.preventDefault();
+    }
+  </script>
+
+  {% if siteKey is defined %}
+    <script src='https://www.google.com/recaptcha/api.js?onload=onRecaptchaLoad&render=explicit' async defer></script>
+
+    <script>
+      function submitForm() {
+        document.querySelector('form').submit();
+      }
+
+      function onRecaptchaLoad() {
+        var loginButton = document.querySelector('button');
+
+        grecaptcha.render(loginButton, {
+          sitekey: '{{ siteKey|e('js')|raw }}',
+          callback: submitForm
+        });
+      }
+
+      ga('send', 'event', 'reCAPTCHA', 'required');
+    </script>
+  {% endif %}
+</head>
+<body class="gradient-bg">
+<div class="mdl-layout mdl-layout--fixed-header fill-viewport">
+  <main class="mdl-layout__content" layout-children="column" child-spacing="center">
+    {% include 'announcement.twig' %}
+
+    <form method="post" autocomplete="off" onsubmit="event.target.onsubmit = preventDefault">
+      <input type="hidden" name="AuthState" value="{{ (stateparams.AuthState??'')|e }}"/>
+
+      {% if csrfToken is defined %}
+        <input type="hidden" name="csrf-token" value="{{ csrfToken|e }}"/>
+      {% endif %}
+
+      <div class="mdl-card mdl-shadow--8dp fill-phone-viewport">
+        <div class="mdl-card__media white-bg margin" layout-children="column">
+          <img src="/logo.png" alt="{{ '{login:logo}'|trans({'%idpName%': idpName}) }}">
+        </div>
+
+        <div class="mdl-card__title center">
+          <h1 class="mdl-card__title-text">
+            {{ '{login:header}'|trans({'%idpName%': idpName}) }}
+          </h1>
+        </div>
+
+        <div class="mdl-card__supporting-text" layout-children="column">
+          <div class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">
+            <label for="username" class="mdl-textfield__label">
+              {{ '{login:label_username}'|trans }}
+            </label>
+
+            <input
+              type="text"
+              required
+              name="username"
+              class="mdl-textfield__input mdl-color-text--black"
+              value="{{ username|e }}"
+              {% if username is empty %} autofocus {% endif %}
+              id="username"
+            >
+          </div>
+
+          <div class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label">
+            <label for="password" class="mdl-textfield__label">
+              {{ '{login:label_password}'|trans }}
+            </label>
+
+            <input
+              type="password"
+              required
+              name="password"
+              class="mdl-textfield__input mdl-color-text--black"
+              {% if not username is empty %} autofocus {% endif %}
```

Comment:
Don't care much either way. I think I prefer readability (change), but emphasis on `not` is fine too (as is).

```suggestion
              {% if username is not empty %} autofocus {% endif %}
```